wd_comp.c

```diff
@@ -768,10 +759,52 @@ static const struct wd_alg_ops wd_comp_ops = {
 int wd_comp_env_init(void)
 {
 	return wd_alg_env_init(&wd_comp_env_config, table,
-			       &wd_comp_ops, ARRAY_SIZE(table));
+			       &wd_comp_ops, ARRAY_SIZE(table), NULL);
 }
 
 void wd_comp_env_uninit(void)
 {
 	return wd_alg_env_uninit(&wd_comp_env_config);
 }
+
+int wd_comp_ctx_num_init(__u32 node, __u32 type, __u32 num, __u8 mode)
+{
+	struct wd_ctx_attr ctx_attr;
+	int ret;
+
+	if (type >= WD_DIR_MAX) {
+		WD_ERR("wrong type(%d))!\n", type);
+		return -WD_EINVAL;
+	}
+
+	ret = wd_set_ctx_attr(&ctx_attr, node, type, mode, num);
+	if (ret)
+		return ret;
+
+	return wd_alg_env_init(&wd_comp_env_config, table,
+			       &wd_comp_ops, ARRAY_SIZE(table), &ctx_attr);
+}
+
+void wd_comp_ctx_num_uninit(void)
+{
+	return wd_alg_env_uninit(&wd_comp_env_config);
+}
+
+int wd_comp_get_env_param(__u32 node, __u32 type, __u32 mode,
+			  __u32 *num, __u8 *is_enable)
```

Comment:
How to use this function? I can't find any code reference to this function.

Reply:
We could use like this, to get decomp async ctx num on node 0.
__u32 num;
__u8 enable;
ret = wd_comp_get_env_param(0, 1, 1, &num, &enable)

Reply:
So it should be invoked by wd_comp driver. Don't add any code without reference. You could append a new patch to use this function.

Reply:
ok,I will add it in test file